docs/data-workflow.md

```diff
@@ -0,0 +1,130 @@
+# Data Files Workflow
+
+Starting with Unicode 15.1, the “source of truth” for not-yet-released versions of
+most of the data files is in
+https://github.com/unicode-org/unicodetools/tree/main/unicodetools/data/ucd/dev
+and parallel .../uca/dev etc. folders.
+
+## UCD/UCA/emoji/idna/security data files
+
+First: Files with a special process.
+
+### Unihan data
+
+List of these files (see https://www.unicode.org/Public/UCD/latest/ucd/):
+*   USourceData.txt
+*   USourceGlyphs.pdf
+*   USourceRSChart.pdf
+*   Unihan.zip
+
+Process:
+*   The “source of truth” is the Unihan database maintained by the CJK/Unihan group, including data maintained by Michel.
+*   The CJK/Unihan group posts data files into an internal location.
+*   KenW vets these files and posts them to https://www.unicode.org/Public/draft/UCD/ucd/ .
+*   A unicodetools GitHub contributor fetches these files, preprocesses the contents of Unihan.zip,
+    and creates a pull request as for “regular” data files.
+    (The processed data files go into .../unicodetools/data/ucd/dev/Unihan.)
+
+### NamesList.txt
+
+*   Iterate between KenW and Michel.
+*   Generated from UnicodeData.txt and an annotations file, using some C program.
+*   Used for generating code charts.
+*   KenW posts NamesList.txt into https://www.unicode.org/Public/draft/UCD/ucd/ .
+*   A unicodetools GitHub contributor fetches this file
+    and creates a pull request as for “regular” data files.
+
+### Folder readmes
+
+The “source of truth” for these is outside of GitHub for now.
+KenW updates or vets these files and posts them to https://www.unicode.org/Public/draft/ .
+A unicodetools GitHub contributor fetches these files and creates a pull request as above.
+
+See https://github.com/unicode-org/properties/issues/8 “simplify versioning of readme files”
+
+### “Regular” data files
+
+Changes are made in a GitHub pull request.
+*   Contributors who don’t use GitHub or don’t contribute to the unicodetools project
+    can send changes (updated files or deltas) to someone who is comfortable using GitHub.
+*   Updated files could be shared in various ways including via email or via private FTP areas.
+*   Updated files should be based on the latest (or fairly recent) data in the unicodetools repo.
+*   Updated files should not be posted directly to https://www.unicode.org/Public/...
+
+Pull request cycle:
+*   One commit for manual or contributed data changes.
+*   Another commit for tool code updates, generated/derived data files,
+    and for fixing test failures (continuous-integration tests or manual tests).
+*   Peer review of the changes.
+*   QA may include pushing files to a private area, e.g., for KenW to review.
+*   More commits as necessary until changes look good, tests pass,
+    and someone (preferably KenW) approves.
+*   **Squash**-and-merge.
```

Comment:
this was probably discussed in the meeting but is there a reason for this when we are being careful to separate generated and non generated commits? Is it for ease of bisection because `git bisect --first-parent` exists now.

It would be nice to be able to quickly skip generated commits.

Reply:
I've foudn it much more useful to spelunk history when generated commits are separate, but I guess that's fine.

Reply:
I suppose that contributors who are both motivated and good with git could squash a messy list of commits into 2 or 3 that make sense to deliberately show progress, but I wouldn't want to *require* fancy footwork.

Reply:
Yeah I think we shouldn't have messy commit lists, it's just nice to not have to deal with generated files in teh same commit

I guess I'm used to rebase-based workflows where you squash your own (git-revise is a very nice tool in this space)

Reply:
I added an option to squash to two commits.